applications/ShapeOptimizationApplication/python_scripts/optimizer_factory.py

```diff
@@ -23,100 +23,32 @@
-    _ValidateConstraintSettings(optimization_settings["constraints"])
-
-# ------------------------------------------------------------------------------
-def _ValidateTopLevelSettings(optimization_settings):
-    default_settings = KM.Parameters("""
-    {
-        "model_settings" : { },
-        "objectives" : [ ],
-        "constraints" : [ ],
-        "design_variables" : { },
-        "optimization_algorithm" : { },
-        "output" : { }
-    }""")
-
-    for key in default_settings.keys():
-        if not optimization_settings.Has(key):
-            raise RuntimeError("CreateOptimizer: Required setting '{}' missing in 'optimization_settings'!".format(key))
-
-    optimization_settings.ValidateAndAssignDefaults(default_settings)
-
-# ------------------------------------------------------------------------------
-def _ValidateObjectiveSettingsRecursively(objective_settings):
-    default_settings = KM.Parameters("""
-    {
-        "identifier"                          : "NO_IDENTIFIER_SPECIFIED",
-        "type"                                : "minimization",
-        "scaling_factor"                      : 1.0,
-        "analyzer"                            : "external",
-        "response_settings"                   : {},
-        "is_combined"                         : false,
-        "combination_type"                    : "sum",
-        "combined_responses"                  : [],
-        "weight"                              : 1.0,
-        "project_gradient_on_surface_normals" : false
-    }""")
-    for itr in range(objective_settings.size()):
-        objective_settings[itr].ValidateAndAssignDefaults(default_settings)
-
-        if objective_settings[itr]["is_combined"].GetBool():
-            _ValidateObjectiveSettingsRecursively(objective_settings[itr]["combined_responses"])
-
-# ------------------------------------------------------------------------------
-def _ValidateConstraintSettings(constraint_settings):
-    default_settings = KM.Parameters("""
-    {
-        "identifier"                          : "NO_IDENTIFIER_SPECIFIED",
-        "type"                                : "<",
-        "scaling_factor"                      : 1.0,
-        "reference"                           : "initial_value",
-        "reference_value"                     : 1.0,
-        "analyzer"                            : "external",
-        "response_settings"                   : {},
-        "project_gradient_on_surface_normals" : false
-    }""")
-    for itr in range(constraint_settings.size()):
-        constraint_settings[itr].ValidateAndAssignDefaults(default_settings)
+## Purely for backward compatibility, should be removed soon.
+def CreateOptimizer(optimization_settings,model,external_analyzer=EmptyAnalyzer()):
+    return Optimizer(model, optimization_settings, external_analyzer)
 
+def Create( model, optimization_settings, external_analyzer=EmptyAnalyzer()):
```

Comment:
Why changing the naming here? I like the changes in this PR, but breaking the existing API just for a minor rewording needs a good argument. I guess it's more "factory-like", right? I am fine with both, though. Just out of curiosity.